### PR TITLE
[Feature] Add dashboard project list tests [OSF-8790]

### PR DESCRIPTION
**Pull request by fabmiz**
## Purpose

Add dashboard project list tests [OSF-8790]

## Changes

- Add encrypted creds for new staging user
- Add pythosf to reqs
- Add tests for project list on the dashboard page

## Ticket

[OSF-8790](https://openscience.atlassian.net/browse/OSF-8790)

### CODE REVIEW

File: api/osf_api.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from pythosf import client
+
+def create_project(session, title='osf selenium test'):
+    node = client.Node(session=session)
+    node.create(title=title)
```
Comment (brianjgeiger) [inline api/osf_api.py]: The next version will likely return the object from the create method, so this should be much simpler in the future. Thank you for the feedback and let me know if you happen to find anything else like that.

File: tests/test_dashboard.py [hunk 2]
```diff
@@ -45,5 +47,78 @@ def test_modal_buttons(self):
...
+
+class TestDashboardPageProjectList(SeleniumTest):
+
+    def setup_method(self, method):
+        self.dashboard_page = DashboardPage(self.driver)
+        self.dashboard_page.goto()
+
+    @pytest.fixture()
+    def project_one(self):
+        project_one = osf.create_project(self.session, title='&&aaaaaa')
```
Comment (brianjgeiger) [inline tests/test_dashboard.py]: Nice ampersands 👍

File: requirements.txt [hunk 3]
```diff
@@ -6,3 +6,4 @@ flake8-quotes==0.3.0
 autopep8==1.3.3
 pytest-xdist==1.15.0
 requests==2.5.3
+pythosf==0.0.3
```
Comment (brianjgeiger) [inline requirements.txt]: Good pinning to the version.

File: tests/test_dashboard.py [hunk 4]
```diff
@@ -45,5 +47,78 @@ def test_modal_buttons(self):
...
+        yield project_one
+        project_one.delete()
+
+    @pytest.fixture()
+    def project_two(self):
+        project_two = osf.create_project(self.session, title='&&aaaabb')
+        yield project_two
+        project_two.delete()
```
Comment (brianjgeiger) [inline tests/test_dashboard.py]: Ooh, that's super handy. I didn't know about the yield-delete pattern with fixtures.

**Comment by brianjgeiger** [comment]
@fabmiz Mind you, it would be better if the tests were passing.

File: tests/test_dashboard.py (hunk 4 above)
Comment (fabmiz) [reply tests/test_dashboard.py]: it's one of my most favorite things to do in pytest. Very convenient. We need to find a way to delete projects created with UI when testing project creation. Maybe run a script daily to clean the test user up.

**Pull request merged**